test/Main.hs

```diff
@@ -79,7 +80,7 @@ main = do
                 , Test.Oscoin.Configuration.tests
                 , Test.Oscoin.Crypto.Blockchain.Eval.tests crypto
                 , Test.Oscoin.Crypto.Blockchain.GeneratorsTest.tests crypto
-                , Test.Oscoin.Crypto.Hash.tests crypto
+                , Test.Oscoin.Crypto.Hash.tests realCrypto
```

Comment:
Why we need this? If you want to exercise the real world crypto imho the correct solution is to leave the crypto generic but to pass `--crypto=realWorld` to your test arguments (see https://github.com/oscoin/oscoin/pull/511). You want this to be run all the time? Then https://github.com/oscoin/oscoin/issues/502 is the answer.

Passing the real crypto here will create a singularity in the tests and we will soon be start adding all this special case around, at which point this would defeat the purpose of having introduced a mock crypto in the first place 😉

Reply:
The thinking was that testing the mock hashing functions is not that useful vs. testing the real hashing functions. Ideally we would test both mock and realworld here. #502 is not really the issue here, because the tests are actually fast to run, and if you're working with the hashing, you actually want to test the realworld crypto.

Reply:
Yeah, I know where you are coming from -- I am just worried that if we start going down this route there is no coming back, we will keep adding a galore of this "special cases" as now it's the hash, in the future it might be anything else crypto-dependent. If we can, I would prefer to keep (one and only) dictionary at the beginning of the test run and use the `--crypto` CLI or, if you are strongly opposed to it, at least a comment describing while are we (deliberately) doing that 😉

Reply:
Yeah, I see the concern - but I think for this suite to be useful, we have to have some kind of compromise, because in this case, what we're testing is actually the `crypto` itself, not something that uses the crypto. Another example would be the tests for the `PubKey` module. So basically what I'm proposing is that for those special cases, we "override" the argument, and test *both* dictionaries. I'll commit something and let me know what you think.

Reply:
I think I can live with such compromise, provided we document those exceptions properly 😉

Reply:
Here it is: c2f04cf